DigitalOffice/Services/UserService/Commands/DisableUserByIdCommand.cs

```diff
@@ -0,0 +1,59 @@
+﻿using LT.DigitalOffice.Broker.Requests;
+using LT.DigitalOffice.Kernel.Broker;
+using LT.DigitalOffice.UserService.Commands.Interfaces;
+using LT.DigitalOffice.UserService.Database.Entities;
+using LT.DigitalOffice.UserService.Repositories.Interfaces;
+using MassTransit;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.UserService.Commands
+{
+    public class DisableUserByIdCommand : IDisableUserByIdCommand
+    {
+        private readonly IUserRepository repository;
+        private readonly IRequestClient<IUserIdRequest> requestClient;
+
+        public DisableUserByIdCommand(
+            [FromServices] IUserRepository repository,
+            [FromServices] IRequestClient<IUserIdRequest> requestClient)
+        {
+            this.repository = repository;
+            this.requestClient = requestClient;
+        }
+
+        public async Task<bool> ExecuteAsync(Guid userId, Guid requestingUserId)
+        {
+            var isUserRight = await GetResultCheckingUserRights(requestingUserId);
+
```

Comment:
1. Change to "doesUserHaveRights" or "hasRights", because it's grammar-correct.
2. You can pass IAccessValidator in the ctor (+ adding .AddKernelExtensions(config) in startup and checking appsettings.json and registering the service in masstransit if it has not yet been registered)
and use it like 
var isAdmin = accessvalidator.IsAdmin()
and then if needed
var hasRights = accessValidator.HasRights(int value);

Reply:
Fixe